benchmark/operations/test/test_create_container.py

```diff
@@ -84,6 +84,43 @@ def cleanup(result):
         # The Deferred fires once the container has been created.
         self.successResultOf(d)
 
+    def test_create_container_timeout(self):
+        """
+        CreateContainer probe times-out if container not created.
+        """
+        clock = Clock()
+
+        node_id = uuid4()
+        node = Node(uuid=node_id, public_address=IPAddress('10.0.0.1'))
+        control_service = FakeFlockerClient([node], node_id)
+
+        cluster = BenchmarkCluster(
+            IPAddress('10.0.0.1'),
+            lambda reactor: control_service,
+            {},
+            None,
+        )
+        operation = CreateContainer(clock, cluster)
+        d = operation.get_probe()
+
+        def run_probe(probe):
+            def cleanup(result):
+                cleaned_up = probe.cleanup()
+                cleaned_up.addCallback(lambda _ignored: result)
+                return cleaned_up
+            d = probe.run()
+            d.addCallback(cleanup)
+            return d
+        d.addCallback(run_probe)
```

Comment:
:question: Since everything is synchronous, I don't think we need to have the cleanup in a callback. We could instead have:

```python
d = operation.get_probe()
def run_probe(probe):
    self.addCleanup(self.successResultOf, probe.cleanup)
    return probe.run()
d.addCallback(run_probe)
```

This has the advantage of clarifying exactly where we expect the failure. I'll make a couple of other comments that relate to this optional suggestion.